i18n/README.md

```diff
@@ -5,11 +5,275 @@
+  - [DuplicatedKeyResolution](#duplicatedkeyresolution)
+  - [Settings](#settings)
+  - [Building](#building)
+- [Translator](#translator)
+  - [Basic usage](#basic-usage)
+  - [Variable interpolation](#variable-interpolation)
+  - [Options](#options)
+- [Configuration](#configuration)
+  - [`interpolationDelimiter`](#interpolationdelimiter)
+  - [`pluralMapper`](#pluralmapper)
+  - [`formats`](#formats)
+  - [`missingKeyPolicy`](#missingkeypolicy)
 - [Changelog](#changelog)
 
 ## Intentions
 
-This module provides a simple way to manage internationalization in a Java/Kotlin project.
+This module provides a simple way to manage internationalization in a Java/Kotlin project, with features such as
+variable interpolation and formatting, and automatic loading of translation files
+
+## TranslatorBuilder
+
+Before anyone can use the `Translator` interface, which will be covered in the next section, one has to first build
+an instance of it.
+
+Thankfully, `TranslatorBuilder` exists for this purpose. The goal is to provide it with different sources of translation
+associations. 
+
+### Representation
+
+Indeed, a translation is a key-value pair, where the key is a `TranslationKey` (which is a type alias for
+`String`), and the value is a `TranslationSourceData` (which is a type alias for `Any`) and corresponds to the
+translation itself. This key-value pair is itself associated to a `Locale`.
+
+In fact, a `TranslationKey` is a `String` that must follow a specific format, and here are some examples of valid keys:
+Here are some examples of valid `TranslationKeys`:
+- `my-root.my-node.my-key`
+- `my_root.my_node.my_key`
+- `myRoot.myNode.myKey`
+- `my-root.my-node.my-table.1`
+
+Any value that does not respect these rules will result in an exception being thrown. For more details on the format of
+`TranslationKeys`, please refer to the documentation of the `TranslationKey` type alias.
+
+A set of those associations is a `TranslationSourceMap`, which is a type alias for 
+`Map<TranslationKey, TranslationSourceData>`.
+
+### Adding translations
+
+To start off, one has to obtain an instance of `TranslatorBuilder` thanks to the `Translator::builder` static method, 
+which takes a default locale as a parameter. This locale will be used as the required locale for the translations.
+
+```kt
+val builder = Translator.builder(Locale.ENGLISH)
+```
+
+The primitive way to add translations is to use the `TranslatorBuilder::addMap` method, which associates a `Locale` to
+a `TranslationSourceMap`. This method can be called multiple times, and the `TranslationSourceMap`s will be merged
+(accordingly to the [`DuplicatedKeyResolution`](#duplicatedkeyresolution) strategy).
+
+```kt
+val builder = Translator.builder(Locale.ENGLISH)
+    .addMap(Locale.ENGLISH, mapOf(
+        "foo" to "My translation",
+        "bar" to "My other translation"
+    ))
+    .addMap(Locale.FRENCH, mapOf(
+        "foo" to "Ma traduction",
+        "bar" to "Mon autre traduction"
+    ))
```

Comment:
For this example and the others, it can be better to build the builder and therefore to change the name of the variable to translator